source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -140,9 +144,26 @@ void AuthenticatorImpl::startVerify() {
   curr_token_ = std::move(tokens_.back());
   tokens_.pop_back();
 
-  jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
-  ENVOY_LOG(debug, "{}: Parse Jwt {}", name(), curr_token_->token());
-  Status status = jwt_->parseFromString(curr_token_->token());
+  if (provider_) {
+    jwks_data_ = jwks_cache_.findByProvider(provider_.value());
+    if (jwks_data_->getJwtProvider().enable_jwt_cache()) {
```

Comment:
enable_jwt_cache should be handled inside JwtCache class   so the code is

```
if (provider_) {
   jwks_data_ = jwks_cache_.findByProvider(provider_.value());
   jwt_ = jwks_data_->getJwtCache().find(curr_token_->token());
   if (jwt_) {
      handleGoodJwt(/*cache_hit=*/true);
      return;
   }
}
```
   }

Reply:
Done.